test/gild/RedeemOverloaded.test.ts

```diff
@@ -0,0 +1,168 @@
+import chai from "chai";
+import { solidity } from "ethereum-waffle";
+import { ethers } from "hardhat";
+import {
+  assertError,
+  deployERC20PriceOracleVault,
+  fixedPointDiv,
+  fixedPointMul,
+  ADDRESS_ZERO,
+  getEventArgs,
+  getEvent,
+} from "../util";
+import { ERC20, ERC20PriceOracleVault } from "../../typechain";
+import { BigNumber } from "ethers";
+import { WithdrawEvent } from "../../typechain/IERC4626";
+
+chai.use(solidity);
+
+const { assert } = chai;
+
+let vault: ERC20PriceOracleVault,
+  asset: ERC20,
+  shareRatio: BigNumber,
+  aliceAddress: string,
+  aliceAssets: BigNumber;
+
+describe("Overloaded Redeem", async function () {
+  beforeEach(async () => {
+    const signers = await ethers.getSigners();
+    const alice = signers[0];
+
+    const [ERC20PriceOracleVault, Erc20Asset, priceOracle] =
+      await deployERC20PriceOracleVault();
+
+    vault = await ERC20PriceOracleVault;
+    asset = await Erc20Asset;
+    shareRatio = await priceOracle.price();
+    aliceAddress = alice.address;
+
+    aliceAssets = ethers.BigNumber.from(5000);
+    await asset.transfer(aliceAddress, aliceAssets);
+
+    await asset.connect(alice).increaseAllowance(vault.address, aliceAssets);
+
+    const depositTx = await vault["deposit(uint256,address,uint256,bytes)"](
+      aliceAssets,
+      aliceAddress,
+      shareRatio,
+      []
+    );
+
+    await depositTx.wait();
+  });
+  it("Redeems", async function () {
```

Comment:
can we please test things other than a user just redeeming all their tokens? what about half their tokens? what about an amount of tokens that rounds up/down weirdly? what about too many tokens?